core/controlplane/config/templates/cloud-config-controller

```diff
@@ -991,7 +991,7 @@ write_files:
           -ec \
           'echo decrypting assets
            shopt -s nullglob
-           for encKey in /etc/kubernetes/{ssl,{{if .AssetsConfig.HasAnyAuthTokens}}auth{{end}}}/*.enc; do
+           for encKey in /etc/kubernetes/{ssl,{{ if or (.AssetsConfig.HasAuthTokens) ( and .Experimental.TLSBootstrap.Enabled .AssetsConfig.HasTLSBootstrapToken) }}auth{{end}}}/*.enc; do
```

Comment:
There would only be one use of ```HasAnyAuthTokens``` (currently on line 680 in config-worker) ```           for encKey in /etc/kubernetes/{ssl,{{if and .Experimental.TLSBootstrap.Enabled .AssetsConfig.HasAnyAuthTokens}}auth{{end}}}/*.enc; do``` I left this alone since I wasn't sure if this logic was meant to be implemented as shown. 
If we can replace the logic used here with ```{{ if or (.AssetsConfig.HasAuthTokens) ( and .Experimental.TLSBootstrap.Enabled .AssetsConfig.HasTLSBootstrapToken) }}```, then we can remove the HasAnyAUthToken method since it would no longer be used.